[Wang Yanxiang] iP

isawangyx opened the pull request:
# ChatterBot
> "A goal without a plan is just a wish." – Antoine de Saint-Exupéry

ChatterBot is your personal task assistant, designed to help you **manage tasks efficiently**. It's: 

- Text-based – Just type your commands!
- Easy to learn – Intuitive commands for task management.
- ~~Fast~~ _Super_ fast! – Stay productive with instant responses.

## Getting Started
1. Download the latest release from [here](https://github.com/isawangyx/ip/releases).
2. Double-click the JAR file or run it using java -jar ChatterBot.jar.
3. Start adding tasks and let ChatterBot handle the rest! 😉

And best of all, it’s **FREE!**

## Features
- Managing tasks ✅
- Managing deadlines ✅
- Finding tasks ✅

## To-Do List 📝
- [x] Implement task management
- [x] Add file storage
- [x] Support finding tasks
- [ ] Implement GUI using JavaFX
- [ ] Add reminders feature

If you're a Java programmer, here’s the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        ChatterBot bot = new ChatterBot("data/chatterbot.txt");
        bot.run();
    }
}
```

💡 ChatterBot is a great way to improve your productivity while practicing Java programming! Give it a try today! 🚀

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yxnxiangg committed:
Rename main class

yxnxiangg committed:
Add greeting

yxnxiangg committed:
Add support for echo

yxnxiangg committed:
Add support for list

yxnxiangg committed:
Add task class

yxnxiangg committed:
Add support for marking task as done and not done

yxnxiangg committed:
Add Deadline, Event and Todo classes

yxnxiangg committed:
Add support for Deadlines, Events and ToDos in main class

yxnxiangg committed:
Add semi-automated testing of ChatterBot with I/O redirection technique

yxnxiangg committed:
Revert "Add semi-automated testing of ChatterBot with I/O redirection technique"

This reverts commit 45bd84c0eaa2d3fce6233530caee0613d718428b.

yxnxiangg committed:
Add Exception classes for handling errors

yxnxiangg committed:
Add support for handling errors in main class

yxnxiangg committed:
Add support for deleting tasks from list

yxnxiangg committed:
Add semi-automated text UI testing using I/O redirection

yxnxiangg committed:
Add Storage class

yxnxiangg committed:
Add support for saving tasks in hard disk

yxnxiangg committed:
Merge Level-7

yxnxiangg committed:
Add support for dates and times

yxnxiangg committed:
Merge Level-8

xk1234 commented on src/main/java/ChatterBot.java:
```diff
@@ -0,0 +1,137 @@
+        storage = new Storage(FILE_PATH);
+        tasks = new ArrayList<>(storage.loadTasks());
+
+        Scanner sc = new Scanner(System.in);
```
Short names can be less clear to readers who are new to the code

xk1234 commented on src/main/java/ChatterBot.java:
```diff
@@ -0,0 +1,137 @@
+        }
+    }
+
+    private static void printAddedTask(Task task, int taskCount) {
```
A well-chosen verb-based name that clearly indicates the method’s purpose

xk1234 commented on src/main/java/ChatterBot.java:
```diff
@@ -0,0 +1,137 @@
+        sc.close();
+    }
+
+    private static void handleListCommand() {
```
Using verbs for methods is helpful for clarifying what the method does

xk1234 reviewed (commented):
LGTM. No issues in general

yxnxiangg committed:
Add Ui class

yxnxiangg committed:
Add Parser class

yxnxiangg committed:
Add TaskList class

yxnxiangg committed:
Refactor ChatterBot for better OOP

yxnxiangg committed:
Merge refactor-oop

yxnxiangg committed:
Organize classes into packages

reaneechua commented on src/main/java/chatterbot/exceptions/ChatterBotException.java:
```diff
@@ -0,0 +1,7 @@
+package chatterbot.exceptions;
+
+public class ChatterBotException extends Exception {
+    public ChatterBotException(String message) {
```
Is this class needed? Is there a reason why you chose to create this class? Seems like this only extends the default Exception class, and when I took a cursory look through your code, it seems like this class is not used(although I might have missed this out)

reaneechua commented on src/main/java/chatterbot/Parser.java:
```diff
@@ -0,0 +1,87 @@
+import chatterbot.exceptions.EmptyDescriptionException;
+import chatterbot.tasks.*;
+
+public class Parser {
```
I really liked the way you wrote the parser! Super clean and easy to read.

reaneechua commented on src/main/java/chatterbot/Parser.java:
```diff
@@ -0,0 +1,87 @@
+
+import chatterbot.exceptions.UnknownCommandException;
+import chatterbot.exceptions.EmptyDescriptionException;
+import chatterbot.tasks.*;
```
Minor detail, if I'm not wrong * shouldn't be used when importing classes.

reaneechua reviewed (commented):
Overall, I think your code is super clean! Well-written and easy to follow your logic. 
I only found 1 instance of breaking the Java convention, besides this your code is done really well. Keep it up!

yxnxiangg committed:
Merge branch 'add-gradle-support'

yxnxiangg committed:
Add JUnit tests

yxnxiangg committed:
Set ChatterBot as mainClass in Gradle build

yxnxiangg committed:
Add JavaDoc comments

yxnxiangg committed:
Add support for finding a task through a keyword

yxnxiangg committed:
Edit the code to comply with Java coding standard

yxnxiangg committed:
Merge branch-A-JavaDoc into master

yxnxiangg committed:
Resolve merge conflicts

yxnxiangg committed:
Resolve merge conflicts

yxnxiangg committed:
Add CheckStyle

yxnxiangg committed:
Merge branch-A-CheckStyle into master

yxnxiangg committed:
Add GUI using plain Java

yxnxiangg committed:
Refactor ChatterBot to use fxml

yxnxiangg committed:
Add Launcher class

yxnxiangg committed:
Add support for handling GUI commands

yxnxiangg committed:
Add support for handling GUI commands

yxnxiangg committed:
Merge branches 'master' and 'master' of https://github.com/isawangyx/ip

* 'master' of https://github.com/isawangyx/ip:
  Add support for handling GUI commands
  Add Launcher class
  Refactor ChatterBot to use fxml
  Add GUI using plain Java

* 'master' of https://github.com/isawangyx/ip:
  Add support for handling GUI commands
  Add Launcher class
  Refactor ChatterBot to use fxml
  Add GUI using plain Java

yxnxiangg committed:
Merge branch 'master' of https://github.com/isawangyx/ip

yxnxiangg committed:
Refactor command handling using enum

yxnxiangg committed:
Add assertions in TaskList and Storage

yxnxiangg committed:
Refactor Task class to improve readability and maintainability

yxnxiangg committed:
Refactor to use Streams API for filtering and printing tasks

isawangyx committed:
Merge pull request #2 from isawangyx/branch-A-Assertions

Add assertions in TaskList and Storage

yxnxiangg committed:
Merge branch 'master' into branch-A-CodeQuality

isawangyx committed:
Merge pull request #3 from isawangyx/branch-A-CodeQuality

Refactor Task class to improve readability and maintainability

yxnxiangg committed:
Merge branch 'master' into branch-A-Streams

isawangyx committed:
Merge pull request #4 from isawangyx/branch-A-Streams

Refactor to use Streams API for filtering and printing tasks

yxnxiangg committed:
Improve unknown command handling to prompt better user input

yxnxiangg committed:
Add support for finding nearest day in which user has a specified free time slot

yxnxiangg committed:
Improve empty description exception handling

yxnxiangg committed:
Merge branch 'branch-B-FindFreeTimes' into master

yxnxiangg committed:
Improve error handling for invalid commands

When a user enters an invalid command, they receive a generic error message without guidance on available commands. This leads to frustration as users may not know how to correct their input.

By providing more informative error messages, users can quickly understand what went wrong and how to fix it.

Let's:
- Improve error messages in UnknownCommandException by suggesting valid commands.
- Improve error messages for EmptyDescriptionException by suggesting valid commands.

This makes the chatbot more user-friendly by providing clear guidance on correct command usage.

yxnxiangg committed:
Fix free time slot to start from current time

The findFreeTime feature was returning free slots based on today's date instead of the current time.

This is incorrect because users expect the search to start from the current time and find the earliest available time slot in the next 7 days.

Let's update the logic to:
- Start the search from `LocalDateTime.now()`, ensuring that the free slot search begins from the exact current time.
- Adjust the working hours check so that available slots respect the defined 8 AM to 10 PM range within each day.

This fix improves user experience by ensuring the feature returns results that align with real-world expectations, making the chatbot's scheduling functionality more intuitive and useful.

yxnxiangg committed:
Enhance GUI: Different colors for user and bot messages in DialogBox

yxnxiangg committed:
Merge branch branch-A-BetterGui into master

yxnxiangg committed:
Add representative screenshot of the product

yxnxiangg committed:
Update user guide

yxnxiangg committed:
Fix GFMD of image in README.md

yxnxiangg committed:
Fix path of screenshot in README.md

yxnxiangg committed:
Fix implementation of deadline

yxnxiangg committed:
Fix Ui issues

isawangyx committed:
Update README.md

isawangyx committed:
Update README.md

isawangyx committed:
Update README.md

yxnxiangg committed:
Fix path of screenshot in README.md
